core/src/main/scala/algebra/lattice/Heyting.scala

```diff
@@ -33,20 +33,10 @@ import scala.{specialized => sp}
  * classical logic, see the boolean algebra type class implemented as
  * `Bool`.
  */
-trait Heyting[@sp(Int, Long) A] extends Any with BoundedDistributiveLattice[A] { self =>
-  def and(a: A, b: A): A
+trait Heyting[@sp(Int, Long) A] extends Any with BoundedDistributiveLattice[A] with Logic[A] { self =>
```

Comment:
similar comment about not breaking compatibility here. Can we make `Heyting` from `Logic`?